sky/skylet/providers/aws/node_provider.py

```diff
@@ -308,15 +308,20 @@ def create_node(self, node_config, tags, count) -> Dict[str, Any]:
             nodes_matching_launch_config = list(
                 self.ec2.instances.filter(Filters=filters_with_launch_config)
             )
+            # launch_time is the latest launch time of the node, rather than the
+            # initial launch time.
             nodes_matching_launch_config.sort(key=lambda n: n.launch_time, reverse=True)
             if len(nodes_matching_launch_config) >= count:
                 reuse_nodes = nodes_matching_launch_config[:count]
             else:
                 nodes_all = list(self.ec2.instances.filter(Filters=filters))
+                nodes_matching_launch_config_ids = {
```

Comment:
nit: prefer to write `set(n.id for n ....)`? `{}` on first glance can be invoke "dict".